From 77ad5324fbad7558dd555cd7cabb0fa22a49e38a Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Wed, 30 Apr 2025 11:06:51 -0500
Subject: [PATCH] sanlock: fix zero io timeout for direct lockspace requests

---
 src/direct.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/direct.c b/src/direct.c
index 208dd3509b14..e009de1a44b5 100644
--- a/src/direct.c
+++ b/src/direct.c
@@ -314,6 +314,8 @@ static int do_delta_action(int action,
 
 	if (!io_timeout)
 		io_timeout = com.io_timeout;
+	if (!io_timeout)
+		io_timeout = DEFAULT_IO_TIMEOUT;
 
 	rv = sizes_from_flags(ls->flags, &sector_size, &align_size, &max_hosts, "LSF");
 	if (rv)
-- 
2.48.1

